source/sigv4.c

```diff
@@ -2455,16 +2485,13 @@ static SigV4Status_t generateCanonicalRequestUntilHeaders( const SigV4Parameters
     const char * pPath = NULL;
     size_t pathLen = 0U;
 
-    if( returnStatus == SigV4Success )
-    {
-        pCanonicalContext->pBufCur = ( char * ) pCanonicalContext->pBufProcessing;
-        pCanonicalContext->bufRemaining = SIGV4_PROCESSING_BUFFER_LENGTH;
+    pCanonicalContext->pBufCur = ( char * ) pCanonicalContext->pBufProcessing;
+    pCanonicalContext->bufRemaining = SIGV4_PROCESSING_BUFFER_LENGTH;
 
-        /* Write the HTTP Request Method to the canonical request. */
-        returnStatus = writeLineToCanonicalRequest( pParams->pHttpParameters->pHttpMethod,
-                                                    pParams->pHttpParameters->httpMethodLen,
-                                                    pCanonicalContext );
-    }
+    /* Write the HTTP Request Method to the canonical request. */
+    returnStatus = writeLineToCanonicalRequest( pParams->pHttpParameters->pHttpMethod,
+                                                pParams->pHttpParameters->httpMethodLen,
+                                                pCanonicalContext );
 
     /* Set defaults for path and algorithm. */
```

Comment:
Return status should be checked here also.

Reply:
Return status is checked on line 2510

Reply:
I think we should check here also for the lines 2470-2481. Because if the status returned line 2492 is not success we don't need to execute the lines 2470-2481.

Reply:
The additional check would be more trouble than its worth since those lines only set two variables that were declared on the function's stack and don't affect the input. I can rearrange those lines to come before this one, but they will still get executed

Reply:
Its better to rearrange those lines.